src/assets/Generator.Shared/NextLinkOperationImplementation.cs

```diff
@@ -208,7 +212,7 @@ internal static bool TryGetApiVersion(Uri startRequestUri, out ReadOnlySpan<char
             // 1.Original request is a put method;
             // 2.Original request is a management plane patch method. For data plane patch method, the final uri will be determinned by the original response header and "_finalStateVia";
             // 3.Original response header contains "Location" and FinalStateVia is configured to OriginalUri.
-            if (_requestMethod == RequestMethod.Put || _requestMethod == RequestMethod.Patch && _startRequestUri.Scheme == "https" && _startRequestUri.Host == "management.azure.com" || _originalResponseHasLocation && _finalStateVia == OperationFinalStateVia.OriginalUri)
+            if (_requestMethod == RequestMethod.Put || _requestMethod == RequestMethod.Patch && (_isMgmtPlane ?? false) || _originalResponseHasLocation && _finalStateVia == OperationFinalStateVia.OriginalUri)
```

Comment:
do we know for sure this behavior only applies to mgmt plane?  It doesn't seem like that should be the case, its ok if dataplane will never use this behavior, but the LRO implementation shouldn't vary for data vs mgmt plane in specification. @annelo-msft  can you comment?

Reply:
I use this to solve [this problem](https://github.com/Azure/autorest.csharp/pull/1720#issue-1062198206) and for now I just want to apply this behavior to mgmt plane. My previous implementation didn't distinguish between mgmt. plane and data plane, which resulted in the test failure in `Azure.Communication.PhoneNumbers`.